lib/service/metric_client_service.dart

```diff
@@ -127,4 +134,36 @@ class MetricClientService {
       mixPanelClient.setLabel(prop, value);
     }
   }
+
+  MixpanelConfig getConfig() {
+    return mixPanelClient.getConfig();
+  }
+
+  Future<void> setConfig(MixpanelConfig config) async {
+    await mixPanelClient.setConfig(config);
+  }
+
+  Future<void> onOpenApp() async {
+    final config = getConfig();
+    final now = DateTime.now();
+    final startDayOfWeek = now.startDayOfWeek;
+    if (startDayOfWeek
+        .isAfter(config.weekStartAt.add(const Duration(days: 7)))) {
+      addEvent(MixpanelEvent.numberUseAppInAWeek, data: {
+        "number": config.countUseAutonomyInWeek,
```

Comment:
We could manipulate the time by directly specifying it.
Here is the document: https://docs.mixpanel.com/docs/tracking/how-tos/events-and-properties